mods/default/functions.lua

```diff
@@ -210,6 +210,61 @@ end
 
 
 --
+-- fence registration helper
+--
+function default.register_fence(name, def)
+	local fence_texture = "default_fence_overlay.png^" .. def.texture ..
+			"^default_fence_overlay.png^[makealpha:255,126,126"
+	def.drawtype = "fencelike"
+	-- allow almost everything to be overridden
+	if not def.inventory_image then
+		def.inventory_image = fence_texture
+	end
+	if not def.wield_image then
+		def.wield_image = fence_texture
+	end
+	if def.texture then
+		def.tiles = { def.texture }
+		def.texture = nil
+	end
+	if not def.paramtype then
+		def.paramtype = "light"
+	end
+	if not def.sunlight_propagates then
+		def.sunlight_propagates = true
+	end
+	if not def.is_ground_content then
+		def.is_ground_content = false
+	end
+	if not def.selection_box then
+		def.selection_box = {
+			type = "fixed",
+			fixed = {-1/7, -1/2, -1/7, 1/7, 1/2, 1/7},
+		}
+	end
+	-- always add to the fence group, even if no group provided
+	if not def.groups then
+		def.groups = {}
+	end
+	if def.groups then
```

Comment:
This is necessarily true, because of the previous block (unless you're doing something screwy like overriding the `def` metatable's `__newindex` of course, which we don't support).